pkg/registry/core/service/portallocator/allocator.go

```diff
@@ -213,3 +256,36 @@ func (r *PortAllocator) contains(port int) (bool, int) {
 func (r *PortAllocator) Destroy() {
 	r.alloc.Destroy()
 }
+
+// EnableMetrics enables metrics recording.
+func (r *PortAllocator) EnableMetrics() {
+	registerMetrics()
+	r.metrics = &metricsRecorder{}
+}
+
+// calculateRangeOffset estimates the offset used on the range for statically allocation based on
+// the following formula `min(max($min, rangeSize/$step), $max)`, described as ~never less than
+// $min or more than $max, with a graduated step function between them~. The function returns 0
+// if any of the parameters is invalid.
+func calculateRangeOffset(pr net.PortRange) int {
+	// default values for min(max($min, rangeSize/$step), $max)
+	const (
+		min  = 16
+		max  = 128
+		step = 32
+	)
+
+	rangeSize := pr.Size
+	if rangeSize < min {
```

Comment:
is not the same case because subnets size are different, but better use <= to avoid possible problems like this

https://github.com/kubernetes/kubernetes/pull/115322

Reply:
It's done.